mixins/imageHelper.ts

```diff
@@ -7,106 +7,178 @@ import Vue from 'vue';
       type: ImageType,
-      element?: HTMLElement,
-      limitByWidth?: boolean
+      id: string,
+      element: HTMLElement
     ) => string;
   }
 }
 
 const imageHelper = Vue.extend({
   methods: {
     /**
-     * Returns the URL of an item's image at a specific size.
+     * Returns the URL of an item's image:
+     * · When 'element' parameter is passed, size of the image will be determined by the element's width & height
+     * · When 'maxWidth' and 'maxHeight' parameters are passed, size of the image will be as requested
+     * · When no 'element' or 'maxWidth' or 'maxHeight' is provided, image will have the original size.
+     *
+     * type - The type of the image to fetch.
+     * item - The item to fetch the image for (optional).
+     * element - The DOM element which size will be used for the image's maximum width or height (optional).
+     * tag - tag of the image to fetch (optional if item is passed).
+     * [itemId=item?.Id] - itemId to get the image from (optional if item is passed).
+     * [maxWidth=element?.clientWidth] - Maximum width of the image (optional).
+     * [maxHeight=element?.clientHeight] - Maximum height of the image (optional).
+     * [quality=90] - Quality level of the image (optional, only relevant for jpeg format).
+     * [limitByWidth=false] - Use the element's width instead of its height for the size calculation.
      *
-     * @param {HTMLElement} element The DOM element which size will be used for the image's maximum width or height.
-     * @param {BaseItemDto} item The item to fetch the image for.
-     * @param {ImageType} type The type of the image to fetch.
-     * @param {boolean} [limitByWidth=false] Use the element's width instead of its height for the size calculation.
+     * @type {getImageUrlForElementParams}
      * @returns {string} The URL for the image, with the base URL set and the options provided.
      */
-    getImageUrlForElement(
-      item: BaseItemDto,
-      type: ImageType,
-      element?: HTMLElement,
+    getImageUrlForElement({
+      type,
+      item,
+      element,
+      tag,
+      itemId = item?.Id,
+      maxWidth = element?.clientWidth,
+      maxHeight = element?.clientHeight,
+      quality = 90,
       limitByWidth = false
-    ): string {
-      // TODO: Refactor this with an options object
-      if (!item) {
-        throw new TypeError('item must not be null or undefined');
-      }
-
-      let itemId;
-      if (item.Type === 'Episode' && type === ImageType.Thumb) {
-        itemId = item.SeriesId;
-        if (item.SeriesThumbImageTag) {
-          type = ImageType.Thumb;
-        } else {
-          type = ImageType.Backdrop;
+    }: getImageUrlForElementParams): string {
+      if (item) {
+        if (!item.ImageTags) {
+          throw new TypeError(
+            'item.ImageTags must not be null or undefined when an item object is passed'
+          );
+        } else if (!tag) {
+          tag = item.ImageTags[type];
```

Comment:
Why do this instead of having `item?.ImageTags?.[type]` as the default value for the `tag` parameter?

Reply:
@MrTimscampi You just solved a thing that was driving me crazy (literally spend hours understanding why) in this PR and in #394!

I was doing ``tag = item?.ImageTags?[type]``

Thank you very much!